classes/log/SubmissionEmailLogDAO.inc.php

```diff
@@ -40,6 +40,39 @@ function getByEventType($submissionId, $eventType, $userId = null) {
 		return parent::_getByEventType(ASSOC_TYPE_SUBMISSION, $submissionId, $eventType, $userId);
 	}
 
+	/**
+	 * Get submission email log entries by submission ID and event type
+	 * @param $submissionId int
+	 * @param $eventType SUBMISSION_EMAIL_...
+	 * @param $userId int optional Return only emails sent to this user.
+	 * @return DAOResultFactory
+	 */
+	function getByEventTypeAndDate($submissionId, $eventType, $userId = null, $date) {
+		$params = array(
+				(int) ASSOC_TYPE_SUBMISSION,
+				(int) $submissionId,
+				(int) $eventType,
+				$date
+			);
+		if ($userId) $params[] = $userId;
```

Comment:
Add a `(int)` cast here too -- it helps keep PostgreSQL happy when invalid data comes in from userland.